design-documents/graph-ql/framework/auth-annotation.md

```diff
@@ -0,0 +1,62 @@
+# Overview
+
+In the scope of adding features to the GraphQL schema, a client like PWA needs to know what queries need authentication.
+This is needed to know to refresh a token each time it expires, so this logic or what queries need a token doesn't get hardcoded into the client side.
+It is also useful to implement the "soft" auth mode for cart, when a token expires, and the client would still store some info from the last sesson client side.
+
+## Annotation
+
+We need add an anotatoin like `@auth` to support metadata that would be needed by the client:
```

Comment:
How about something that will allow extension for B2B use cases, and potentially admin use cases like staging preview:

```
directive @auth(
  requires: UserRole = ANONYMOUS_USER,
) on OBJECT | FIELD_DEFINITION

enum UserRole {
  REGISTERED_CUSTOMER
  ANONYMOUS_USER
  COMPANY_USER
  ADMIN_USER
}
```
With such schema we can also add additional argument to the `@auth` directive listing required B2B or Admin ACL permissions.

Reply:
ok, I'll update the document